pygfx/_version.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+# This file is part of 'miniver': https://github.com/jbweston/miniver
+#
+from collections import namedtuple
+import os
+
+Version = namedtuple("Version", ("release", "post", "labels"))
+
+# No public API
+__all__ = []
+
+package_root = os.path.dirname(os.path.realpath(__file__))
+package_name = os.path.basename(package_root)
+STATIC_VERSION_FILE = "_static_version.py"
+
+
+def get_version():
+    version = get_version_from_git()
+    if not version:
+        version = Version("unknown", None, None)
+    return pep440_format(version)
+
+
+def pep440_format(version_info):
+    release, post, labels = version_info
+
+    version_parts = [release]
+    if post:
+        if release.endswith("-post") or release.endswith(".post"):
+            version_parts.append(post)
+        else:  # prefer PEP440 over strict adhesion to semver
+            version_parts.append(".post{}".format(post))
+
+    if labels:
+        version_parts.append("+")
+        version_parts.append(".".join(labels))
+
+    return "".join(version_parts)
+
+
+def get_version_from_git():
+    import subprocess
+
+    # git describe --first-parent does not take into account tags from branches
+    # that were merged-in. The '--long' flag gets us the 'post' version and
+    # git hash, '--always' returns the git hash even if there are no tags.
+    for opts in [["--first-parent"], []]:
+        try:
+            p = subprocess.Popen(
+                ["git", "describe", "--long", "--always", "--tags", "--dirty"] + opts,
```

Comment:
IIUC this means that if someone obtains pygfx in another way than via a package manager or git repo, the version cannot be established.

Reply:
Correct. There become some "edge cases" that you have to start taking into account when people "receive" the source code. 

I just checked my project for example, the version downloaded from pypi burns things in due to the modifications to the setup.py but the archive from github would make it pretty difficult for people to use.

![image](https://github.com/pygfx/pygfx/assets/90008/9ac516f9-9e35-4e05-864e-f04eb0e70696)